src/com/owncloud/android/files/FileOperationsHelper.java

```diff
@@ -465,10 +470,23 @@ public void toggleFavorite(OCFile file, boolean isFavorite) {
         mFileActivity.startService(observedFileIntent);
 
         /// immediate content synchronization
-        if (file.isFavorite()) {
+        if (file.getFavoriteStatus() == OCFile.FavoriteStatus.FAVORITE.getValue()) {
             syncFile(file);
         }
     }
+
+    private void toggleAvailableOfflineFilesInFolder(OCFile file, boolean isAvailableOffline) {
+        int favoriteStatus = isAvailableOffline ?
+                OCFile.FavoriteStatus.FAVORITE.getValue() : OCFile.FavoriteStatus.NO_FAVORITE.getValue();
+        Vector<OCFile> filesInFolder = mFileActivity.getStorageManager().getFolderContent(file);
```

Comment:
This might be hard in performance if the subtree under ```file``` is really big. We should try to change the property for any descendant minimizing database access.